global/indico_global/cli.py

```diff
@@ -0,0 +1,112 @@
+from indico.core.settings import SettingsProxyBase
+from indico.modules.categories import Category
+from indico.modules.events import Event
+from indico.util.console import verbose_iterator
+
+from indico_global.models.id_map import GlobalIdMap
+
+
+@cli_group(name='global')
+def cli():
+    """Manage the Global plugin."""
+
+
+@cli.command()
+@click.argument('mapping_file', type=click.File())
+def load_mapping(mapping_file):
+    """Import the ID mapping from YAML."""
+    if GlobalIdMap.query.has_rows():
+        click.secho('Mapping table is not empty', fg='yellow')
+        if not click.confirm('Continue anyway?'):
+            sys.exit(1)
+
+    click.echo('Loading mapping data (this take take a while)...')
+    mapping = yaml.safe_load(mapping_file)
+    for col, data in mapping.items():
+        click.echo(f'Processing {col}...')
+        for local_id, global_id in verbose_iterator(data.items(), len(data), get_id=itemgetter(0)):
+            GlobalIdMap.create(col, local_id, global_id)
+
+    click.echo('Import finished, committing data...')
+    db.session.commit()
+
+
+@cli.command()
+def notify_category_managers():
+    """Notify category managers about upcoming migration."""
+    from indico_global.plugin import GlobalPlugin
+
+    SettingsProxyBase.allow_cache_outside_request = True  # avoid re-querying site_title for every email
+    global_cat = Category.get(GlobalPlugin.settings.get('global_category_id'))
+    query = (global_cat.deep_children_query
+            .filter(~Category.is_deleted, Category.acl_entries.any())
+            .options(subqueryload(Category.acl_entries), undefer('chain_titles')))
+    managers = defaultdict(set)
+    for cat in query:
+        if not (cat_managers := {x.user for x in cat.acl_entries if x.full_access and x.type == PrincipalType.user}):
+            continue
+        for user in cat_managers:
+            managers[user].add(cat)
+
+    for user, cats in managers.items():
+        group_acls = {
+            x.multipass_group_name
+            for cat in cats
+            for x in cat.acl_entries if x.type == PrincipalType.multipass_group
+        }
+        tpl = get_plugin_template_module('emails/cat_notification.txt', name=user.first_name, categories=cats,
+                                         group_acls=group_acls, reply_to='indico-team@cern.ch')
+        send_email(make_email(to_list={user.email}, template=tpl))
+
+
+@cli.command()
+def notify_event_managers():
+    """Notify event managers about upcoming migration."""
+    from indico_global.plugin import GlobalPlugin
+
+    SettingsProxyBase.allow_cache_outside_request = True  # avoid re-querying site_title for every email
+    global_cat = Category.get(GlobalPlugin.settings.get('global_category_id'))
+    query = (Event.query
+            .filter(Event.category_chain_overlaps(global_cat.id),
+                    ~Event.is_deleted,
+                    Event.acl_entries.any(),
+                    Event.end_dt >= datetime(2024, 7, 1))
```

Comment:
Not sure yet whether to include some past events or really just use future events and maybe those that happened in 2025...

Reply:
Maybe last 6/12 months? Depending on how many more e-mails that would be..

Reply:
it would add a lot:

```
In [3]: global_cat = Category.get(GlobalRedirectPlugin.settings.get('global_category_id'))

In [4]:     query = (Event.query
   ...:             .filter(Event.category_chain_overlaps(global_cat.id),
   ...:                     ~Event.is_deleted,
   ...:                     Event.acl_entries.any(),
   ...:                     Event.start_dt >= datetime(2025, 1, 1))
   ...:             .options(subqueryload(Category.acl_entries), undefer('chain_titles'))); query.count()
Out[4]: 157

In [5]:

In [5]:     query = (Event.query
   ...:             .filter(Event.category_chain_overlaps(global_cat.id),
   ...:                     ~Event.is_deleted,
   ...:                     Event.acl_entries.any(),
   ...:                     Event.start_dt >= datetime(2024, 1, 1))
   ...:             .options(subqueryload(Category.acl_entries), undefer('chain_titles'))); query.count()
Out[5]: 2678

In [6]:     query = (Event.query
   ...:             .filter(Event.category_chain_overlaps(global_cat.id),
   ...:                     ~Event.is_deleted,
   ...:                     Event.acl_entries.any(),
   ...:                     Event.start_dt >= datetime(2024, 7, 1))
   ...:             .options(subqueryload(Category.acl_entries), undefer('chain_titles'))); query.count()
Out[6]: 1343

In [7]:     query = (Event.query
   ...:             .filter(Event.category_chain_overlaps(global_cat.id),
   ...:                     ~Event.is_deleted,
   ...:                     Event.acl_entries.any(),
   ...:                     Event.start_dt >= datetime(2024, 12, 1))
   ...:             .options(subqueryload(Category.acl_entries), undefer('chain_titles'))); query.count()
Out[7]: 265

In [8]:     query = (Event.query
   ...:             .filter(Event.category_chain_overlaps(global_cat.id),
   ...:                     ~Event.is_deleted,
   ...:                     Event.acl_entries.any(),
   ...:                     Event.start_dt >= datetime(2025, 1, 1))
   ...:             .options(subqueryload(Category.acl_entries), undefer('chain_titles'))); query.count()
Out[8]: 157
```